_posts/2016-08-02-what-does-RESTful-really-mean.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+asset-type: post
+name: What-does-RESTful-really-mean
+title: 'What does RESTful really mean'
+date: 2016-08-02 17:00:00 +00:00
+author: Carlos Morera de la Chica
+image:
+    src: /assets/img/custom/blog/RESTful.png
+tags:
+- REST
+- Web API
+- architecture
+
+---
+
+Last month I attended [Fast Track to RESTful Microservices training](https://skillsmatter.com/courses/541-fast-track-to-restful-to-microservices) at [Skillsmatter](https://skillsmatter.com). During the course, we explored what REST APIs can offer to web applications in general and microservices communication in particular. Personally, the most important outcome of the course is a better understanding of what REST really means and what are its pros and cons.
+
+During most of my career, I've been focused on mobile technologies, therefore, being on the consuming side of web APIs. Most APIs that I've ever consumed were supposed to be RESTful, now that I better understand what RESTful means I can say that 99%  of them weren't even close to RESTful (Clients are broken because of something being added in the server side sounds familiar?).
+
+## Definition
+
+The term REST stands for "REpresentational State Transfer". A possible formal definition of it could be as follows.
+
+> Architectural API style based on transferring a representation of state (documents) between client and server in order to progress the state of an application.
+
+## Constraints
+
+To consider applications as RESTful, applications need to conform to the following REST constraints. Complying with the constraints enables a distributed hypermedia system to have the following desirable non-functional properties: performance, scalability, simplicity, modifiability, visibility, portability and reliability.
+
+* Client-server
+
+A client-server model induces separation of concerns so that clients are not concerned with data storage. Thus, clients code *portability* is improved. On the other hand, the server is not concerned about user interface or user state, so that server can be *simpler* and more *scalable*. Servers and clients can be developed independently, as long as they conform to the defined contract.
+
+* Stateless
+
+Client context is never stored on the server between requests. Each request has to contain all the necessary information. A stateless server improves *scalability* by allowing the server to quickly free resources and simplifies implementation, *reliability* eases recovering from partial failures, *visibility* Monitoring system does not have to look beyond a single request to determine the nature of the request.
```

Comment:
Second sentence is pretty long, even with the commas. At the same time Monitoring is uppercased.